### ISSUE TITLE
Test execution speed degraded

### ISSUE THREAD

**Issue by Davindroid** [microsoft/WinAppDriver#935]
I am working on Visual Studio 2019, building TestFramework for WPF application. Everything on my script was fine until some recent updates in the application by developers. As a result now my script is taking almost 20x more time to complete test cases.
> My tests are to click on 20 buttons one by one,
Basically i have two versions of my test application say 1 and 2.

On version 1 WinAppDriver takes 25 seconds to run 20 tests
On version 2 WinAppDriver takes 8-11 minutes to run same 20 tests. (some test timeouts)

To mention, i can manually click on all the buttons with no delay or timeouts in version#2 . How can i troubleshoot this ?

WinAppDriver listening on 127.0.0.1 :4723
The only change i am observing in version 2 is this line below appears: 
{"using":"name","value":"NewButton"}  ---- big delay after this and then click happens
If i use version 1, this line doesn't even show up and no delay. everything works fine.

**Comment by hassanuz** [comment]
How are you identifying the button? Is it through the AutomationID or XPath?

**Comment by Davindroid** [comment]
Hi @hassanuz , in my script finding most of buttons by Name buty some of them are using AutomationID as well. we are currently working on assigning Automation ID to elements.

I am using UI recorder to find Xpath of the button and then filtering out the Name/AutomationID what ever matches with inpect.exe parameters.

**Comment by Davindroid** [comment]
This is what we found :
"Delay was caused because the enabled state of a button would go into a feedback loop where it would evaluated its enabled state, and as a result would cause it to re-evaluate its enabled state repeatedly.  This was not enough to hang the UI, but instead cause it to act slowly or inconsistently.

that is the reason why i was able to click manually but script was failing. so it was definitely something in my test application. 

Closing this issue now
